From fb66bd9877c9732a18a03ac90a2ae500ffc3a0d0 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Tue, 17 Sep 2024 13:28:58 +0200
Subject: [PATCH 3/6] build: get rid of {properties,src}/libutils.la

Useless build of an extra libraries, just making the build slower and
more complicated. Get rid of then, and just roll src/libutils.la.
---
 Makefile.am | 69 +++++++++++++++++++++--------------------------------
 1 file changed, 27 insertions(+), 42 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 29084a9..d46cfcd 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -33,6 +33,26 @@ nmvpnservice_DATA = nm-libreswan-service.name
 
 ###############################################################################
 
+noinst_LTLIBRARIES += shared/libutils.la
+
+shared_libutils_la_SOURCES = \
+	shared/nm-utils/nm-shared-utils.c \
+	shared/nm-utils/nm-shared-utils.h \
+	shared/utils.c \
+	shared/utils.h \
+	shared/nm-service-defines.h
+
+shared_libutils_la_CFLAGS = \
+	-DPREFIX=\""$(prefix)"\" \
+	$(common_CFLAGS) \
+	$(LIBNM_CFLAGS)
+
+shared_libutils_la_LIBADD = \
+	$(GLIB_LIBS) \
+	$(LIBNM_LIBS)
+
+###############################################################################
+
 properties/resources.h: properties/gresource.xml
 	$(AM_V_GEN) $(GLIB_COMPILE_RESOURCES) $< --target=$@ --sourcedir=$(srcdir)/properties --generate-header --internal
 
@@ -53,10 +73,6 @@ gtk4/%.ui: properties/%.ui
 EXTRA_DIST += \
 	gtk4/nm-libreswan-dialog.ui
 
-plugin_sources = \
-	properties/nm-libreswan-editor-plugin.c \
-	properties/nm-libreswan-editor-plugin.h
-
 editor_sources = \
 	properties/nm-libreswan-editor.c \
 	properties/nm-libreswan-editor.h
@@ -68,23 +84,6 @@ common_CFLAGS = \
 
 ###############################################################################
 
-noinst_LTLIBRARIES += properties/libutils.la
-
-properties_libutils_la_SOURCES = \
-	shared/utils.c \
-	shared/utils.h \
-	shared/nm-utils/nm-vpn-plugin-utils.c \
-	shared/nm-utils/nm-vpn-plugin-utils.h \
-	shared/nm-utils/nm-shared-utils.c \
-	shared/nm-utils/nm-shared-utils.h \
-	shared/nm-service-defines.h
-
-properties_libutils_la_CPPFLAGS = \
-	-DPREFIX=\""$(prefix)"\" \
-	-DNETWORKMANAGER_COMPILATION=NM_NETWORKMANAGER_COMPILATION_LIB_BASE \
-	$(common_CFLAGS) \
-	$(LIBNM_CFLAGS)
-
 plugin_LTLIBRARIES += properties/libnm-vpn-plugin-libreswan.la
 
 properties_libnm_vpn_plugin_libreswan_la_CFLAGS = \
@@ -93,10 +92,13 @@ properties_libnm_vpn_plugin_libreswan_la_CFLAGS = \
 	$(LIBNM_CFLAGS)
 
 properties_libnm_vpn_plugin_libreswan_la_SOURCES = \
-	$(plugin_sources)
+	shared/nm-utils/nm-vpn-plugin-utils.c \
+	shared/nm-utils/nm-vpn-plugin-utils.h \
+	properties/nm-libreswan-editor-plugin.c \
+	properties/nm-libreswan-editor-plugin.h
 
 properties_libnm_vpn_plugin_libreswan_la_LIBADD = \
-	properties/libutils.la \
+	shared/libutils.la \
 	$(LIBNM_LIBS) \
 	$(DL_LIBS)
 
@@ -198,7 +200,6 @@ auth_dialog_nm_libreswan_auth_dialog_LDADD = \
 
 src_cppflags = \
 	-DBINDIR=\"$(bindir)\" \
-	-DPREFIX=\""$(prefix)"\" \
 	-DLIBDIR=\""$(libdir)"\" \
 	-DLIBEXECDIR=\""$(libexecdir)"\" \
 	-DLOCALSTATEDIR=\""$(localstatedir)"\" \
@@ -230,22 +231,6 @@ src/nm-libreswan-helper-service-dbus.h: src/nm-libreswan-helper-service.xml
 src/nm-libreswan-helper-service-dbus.c: src/nm-libreswan-helper-service-dbus.h
 	@true
 
-noinst_LTLIBRARIES += src/libutils.la
-
-src_libutils_la_SOURCES = \
-	shared/nm-utils/nm-shared-utils.c \
-	shared/nm-utils/nm-shared-utils.h \
-	shared/utils.c \
-	shared/utils.h \
-	shared/nm-service-defines.h
-
-src_libutils_la_CPPFLAGS = \
-	$(src_cppflags)
-
-src_libutils_la_LIBADD = \
-	$(GLIB_LIBS) \
-	$(LIBNM_LIBS)
-
 ###############################################################################
 
 libexec_PROGRAMS += src/nm-libreswan-service
@@ -255,7 +240,7 @@ src_nm_libreswan_service_CPPFLAGS = \
 
 src_nm_libreswan_service_LDADD = \
 	src/libnm-libreswan-helper-service-dbus.la \
-	src/libutils.la \
+	shared/libutils.la \
 	$(GLIB_LIBS) \
 	$(LIBNM_LIBS) \
 	$(LIBNL_LIBS) \
@@ -272,7 +257,7 @@ src_nm_libreswan_service_helper_CPPFLAGS = \
 
 src_nm_libreswan_service_helper_LDADD = \
 	src/libnm-libreswan-helper-service-dbus.la \
-	src/libutils.la \
+	shared/libutils.la \
 	$(GLIB_LIBS) \
 	$(LIBNM_LIBS)
 
-- 
2.47.0

